subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -460,5 +472,37 @@ public FileCollection getAnnotationProcessorPath() {
     public void setAnnotationProcessorPath(@Nullable FileCollection annotationProcessorPath) {
         this.annotationProcessorPath = annotationProcessorPath;
     }
-}
 
+    /**
+     * Returns the directory to place source files generated by annotation compilers.
+     *
+     * @since 4.3
+     */
+    @Optional
+    @Nullable
+    @Incubating
+    @OutputDirectory
+    public File getAnnotationProcessorGeneratedSourcesDirectory() {
+        return annotationProcessorGeneratedSourcesDirectory.getOrNull();
+    }
+
+    /**
+     * Sets the directory to place source files generated by annotation compilers.
+     *
+     * @since 4.3
+     */
+    @Incubating
+    public void setAnnotationProcessorGeneratedSourcesDirectory(File file) {
```

Comment:
Let's not add a setter, just a getter for `DirectoryVar`.

Reply:
Won't use `DirectoryVar` for now.